### PR TITLE
SISRP-50100: Tasks Card - bCourses retrofit

### PR DESCRIPTION
**Pull request by developish**
https://jira-secure.berkeley.edu/browse/SISRP-50100

### CODE REVIEW

**Comment by developish** [comment]
Please DO NOT MERGE until I've also reviewed

File: app/models/user/canvas/activity.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          url: html_url
+        }
+      end
+
+      def initialize(attrs={})
+        attrs.each do |key, value|
+          if respond_to?("#{key}=")
+            send("#{key}=", value)
+          else
+            Rails.logger.debug("- - - - - - - - - - -didn't set #{key}")
```
Comment (marcpechaitis) [inline app/models/user/canvas/activity.rb]: @developish are you leaving this logger statement in intentionally?

File: src/react/components/_dashboard/TasksCard/IncompleteTasks/BCoursesTasks.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+//     hasDetail={false}
+//   >
+//     <TaskHeader task={task}>
+//       <CampusSolutionsIcon />
+//       <TaskTitle title={task.title} />
+
+//       {task.url && <APILink {...task.url} name={<ViewLink />} />}
+//     </TaskHeader>
+//   </Task>
+// );
```
Comment (marcpechaitis) [inline src/react/components/_dashboard/TasksCard/IncompleteTasks/BCoursesTasks.js]: @developish delete this commented out code?

File: src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js [hunk 3]
```diff
@@ -42,12 +43,16 @@ const ByCategory = ({ categories }) => {
               aidYears={category.aidYears}
             />
           );
+        } else if (category.key === 'bCourses' && category.tasks.length > 0) {
+          return <BCoursesTasks key={category.key} tasks={category.tasks} />;
         } else if (category.tasks.length > 0) {
           return <ForCategory key={category.key} category={category} />;
```
Comment (marcpechaitis) [inline src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js]: @developish could this be refactored?

```
} else if (category.tasks.length >0 {
  category.key === 'bCourses' ?
    return <BCoursesTasks key={category.key} tasks={category.tasks} />;
  :
    return <ForCategory key={category.key} category={category} />;
}
```

File: src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js [hunk 4]
```diff
@@ -42,12 +43,16 @@ const ByCategory = ({ categories }) => {
...
         }
 
         return null;
       })}
+
+      {/* <BCoursesTasks /> */}
```
Comment (marcpechaitis) [inline src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js]: @developish remove commented code?

File: src/react/components/_dashboard/TasksCard/tasks.module.js [hunk 5]
```diff
@@ -86,3 +90,12 @@ export const groupByCategory = (accumulator, value) => {
 
   return accumulator;
 };
+
+import { isWithinInterval, parseISO, addDays } from 'date-fns';
+import { startOfToday } from 'date-fns';
```
Comment (marcpechaitis) [inline src/react/components/_dashboard/TasksCard/tasks.module.js]: @developish combine into one line?

File: src/react/components/_finances/AidYearTaskCount.js [hunk 6]
```diff
@@ -61,8 +61,13 @@ const mapStateToProps = ({
   myAgreements: { incompleteAgreements = [], loaded: agreementLoaded },
   myChecklistItems: { incompleteItems = [], loaded: checklistLoaded },
 }) => {
+<<<<<<< HEAD
   const tasks = [...incompleteAgreements, ...incompleteItems].filter(task => {
     return task.displayCategory === 'financialAid' && !task.isBeingProcessed;
+=======
+  const tasks = [...activeAgreements, ...incompleteItems].filter(task => {
+    return task.displayCategory === 'financialAid';
+>>>>>>> SISRP-50100: Tasks Card - bCourses retrofit
```
Comment (marcpechaitis) [inline src/react/components/_finances/AidYearTaskCount.js]: @developish incomplete merge conflicts?

File: src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js (hunk 3 above)
Comment (developish) [reply src/react/components/_dashboard/TasksCard/IncompleteTasks/ByCategory.js]: I'd would have disagreed on the small refactor, but I refactored the whole thing to a wrapper conditional on length plus a switch on category.key

File: app/models/user/canvas/activity.rb (hunk 1 above)
Comment (developish) [reply app/models/user/canvas/activity.rb]: Nope

**Pull request merged**